controllers/componentdetectionquery_controller_test.go

```diff
@@ -1205,6 +1205,56 @@ var _ = Describe("Component Detection Query controller", func() {
 				deleteCompDetQueryCR(hasCompDetQueryLookupKey)
 			})
 		})
+
+		Context("Create Component Detection Query for spring boot repository with multiple components detected, one at root", func() {
+			It("Should only return one component, a spring boot devfile", func() {
+				ctx := context.Background()
+
+				queryName := "spring-boot-root-component"
+
+				hasCompDetectionQuery := &appstudiov1alpha1.ComponentDetectionQuery{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "appstudio.redhat.com/v1alpha1",
+						Kind:       "ComponentDetectionQuery",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      queryName,
+						Namespace: HASNamespace,
+					},
+					Spec: appstudiov1alpha1.ComponentDetectionQuerySpec{
+						GitSource: appstudiov1alpha1.GitSource{
+							URL:      "https://github.com/devfile-resources/todo-spring-boot",
+							Revision: "main",
+						},
+					},
+				}
+
+				Expect(k8sClient.Create(ctx, hasCompDetectionQuery)).Should(Succeed())
+
+				// Look up the has app resource that was created.
+				// num(conditions) may still be < 1 on the first try, so retry until at least _some_ condition is set
+				hasCompDetQueryLookupKey := types.NamespacedName{Name: queryName, Namespace: HASNamespace}
+				createdHasCompDetectionQuery := &appstudiov1alpha1.ComponentDetectionQuery{}
+				Eventually(func() bool {
+					k8sClient.Get(context.Background(), hasCompDetQueryLookupKey, createdHasCompDetectionQuery)
+					return len(createdHasCompDetectionQuery.Status.Conditions) > 1
+				}, timeout, interval).Should(BeTrue())
+
+				// Make sure the a devfile is detected
+				Expect(len(createdHasCompDetectionQuery.Status.ComponentDetected)).Should(Equal(1))
+
+				for devfileName, devfileDesc := range createdHasCompDetectionQuery.Status.ComponentDetected {
+					Expect(devfileName).Should(ContainSubstring("spring"))
+					Expect(devfileDesc.ComponentStub.Source.GitSource.Context).Should(ContainSubstring("./"))
+					Expect(devfileDesc.ComponentStub.Source.GitSource.Revision).Should(ContainSubstring("main"))
+					Expect(devfileDesc.ComponentStub.Source.GitSource.DevfileURL).Should(Equal("https://registry.stage.devfile.io/devfiles/java-springboot-basic"))
```

Comment:
To fix this issue: https://issues.redhat.com/browse/DEVHAS-249
all the `devfileURL` will be returned with the github raw file url instead of the one from registry.

Reply:
Thanks Stephanie! Updated 🙂